src/net/net.c

```diff
@@ -95,6 +95,22 @@ int net_dst_source_addr_get(struct sa *dst, struct sa *ip)
 }
 
 
+int net_dst_is_source_addr(const struct sa *dst, const struct sa *ip)
+{
+	struct sa src;
+	int err;
+
+	err = net_dst_source_addr_get(dst, &src);
+	if (err)
+		return err;
+
+	if (!sa_cmp(ip, &src, SA_ADDR))
+		return ECONNREFUSED;
+
+	return 0;
+}
```

Comment:
I dont think that this function belongs in a library.

it is just comparing one IP with another IP. this can be done in the application